packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -113,18 +114,16 @@ const LinkControlSearchInput = forwardRef(
 			}
 		};
 
-		const inputLabel = placeholder ?? __( 'Search or type URL' );
-
 		return (
 			<div className="block-editor-link-control__search-input-container">
 				<URLInput
 					disableSuggestions={ currentLink?.url === value }
-					label={ inputLabel }
+					label={ label ?? __( 'Search or type URL' ) }
 					hideLabelFromVision={ hideLabelFromVision }
 					className={ className }
 					value={ value }
 					onChange={ onInputChange }
-					placeholder={ inputLabel }
+					placeholder={ placeholder }
```

Comment:
From the [**Why** section of the PR that introduced the duplicative change](https://github.com/WordPress/gutenberg/pull/65458), the rationale is that if there is no visible label then the placeholder text must match the label. It doesn't look like `<InputBase />` handles this for us, so we could do it here:

1. Set a const for the inputLabel above this block: `const inputLabel =  label ?? __( 'Search or type URL' )`
2. Apply the label: `label={ inputLabel }`
3. Check for hideLabelFromVision before applying the placeholder: `placeholder={ hideLabelFromVision ? inputLabel : placeholder }`

Reply:
👍 We don't want to end up in a situation where we have the `label` and `placeholder` showing the same thing. That's the principle bug we have to address in this PR. Visual reference:

![Image](https://github.com/user-attachments/assets/714fae15-0163-4787-bb73-4ccc1a7c16ed)

Reply:
Yup - hopefully the sample code I added will:
- Not have the placeholder and label match unless the consumer passes strings that match.
- Force the placeholder to match the label _only if hideLabelFromVision is true_ and we would want the placeholder and label to match.